src/ApiService/ApiService/onefuzzlib/orm/Orm.cs

```diff
@@ -134,6 +141,57 @@ public IAsyncEnumerable<T> SearchByRowKeys(IEnumerable<string> rowKeys)
         public IAsyncEnumerable<T> SearchByTimeRange(DateTimeOffset min, DateTimeOffset max) {
             return QueryAsync(Query.TimeRange(min, max));
         }
+
+        public async Task<List<ResultVoid<(int, string)>>> BatchOperation(IAsyncEnumerable<T> entities, TableTransactionActionType actionType) {
+            var tableClient = await GetTableClient(typeof(T).Name);
+            var transactions = await entities.Select(e => new TableTransactionAction(actionType, _entityConverter.ToTableEntity(e))).ToListAsync();
+            var responses = await tableClient.SubmitTransactionAsync(transactions);
+            return responses.Value.Select(response =>
+                response.IsError ? ResultVoid<(int, string)>.Error((response.Status, response.ReasonPhrase)) : ResultVoid<(int, string)>.Ok()
+            ).ToList();
+        }
+
+
+        public async Task<DeleteAllResult> DeleteAll(IEnumerable<(string?, string?)> keys) {
+            var query = Query.Or(
+                keys.Select(key =>
+                    key switch {
+                        (null, null) => throw new ArgumentException("partitionKey and rowKey cannot both be null"),
+                        (string partitionKey, null) => Query.PartitionKey(partitionKey),
+                        (null, string rowKey) => Query.RowKey(rowKey),
+                        (string partitionKey, string rowKey) => Query.And(
+                            Query.PartitionKey(partitionKey),
+                            Query.RowKey(rowKey)
+                        ),
+                    }
+                )
+            );
+
+            var tableClient = await GetTableClient(typeof(T).Name);
+            var pages = tableClient.QueryAsync<TableEntity>(query, select: new[] { "PartitionKey, RowKey" });
+
+            var requests = await pages
+                .Chunk(MAX_TRANSACTION_SIZE)
+                .Where(chunk => chunk.Count > 0)
+                .Select(chunk =>
+                    chunk.Select(e => new TableTransactionAction(TableTransactionActionType.Delete, e))
+                )
+                .Select(t => tableClient.SubmitTransactionAsync(t))
+                .ToArrayAsync();
```

Comment:
small, but using `ToList` does less copying if we don't need an array
```suggestion
                .ToListAsync();
```